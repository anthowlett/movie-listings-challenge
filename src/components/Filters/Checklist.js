import React from 'react';
import PropTypes from 'prop-types';
import Checkbox from './Checkbox';
import styled from 'styled-components';
import { colors } from '../../utils';

// Styles
const Message = styled.p`
  padding: 5px;
  color: red;
  background-color: ${colors.black_5};
`;

const Checklist = ({
	list,
	checked,
	onChange
}) => (
  <div>
    {
      !list.length ? 
        <Message>Opps! No movies means no filters :/</Message> : 
        list
          .sort((a, b) => a.name.localeCompare(b.name))
          .map((item) => {
            return (
              <Checkbox key={item.id} id={item.id} name={item.name} checked={checked.includes(item.id)} onChange={onChange} />
            );
          })
    }
  </div>
);

Checklist.propTypes = {
    list: PropTypes.array.isRequired,
    checked: PropTypes.array.isRequired,
    onChange: PropTypes.func.isRequired
};

export default Checklist;