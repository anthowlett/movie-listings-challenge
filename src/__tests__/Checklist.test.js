import React from 'react';
import renderer from 'react-test-renderer';
import Checklist from '../components/Filters/Checklist';
import '../setupTests';

describe('Filters', () => {
  
  describe('<Checklist />', () => {
    const genres = [
      {id: 28, name: "Action"},
      {id: 12, name: "Adventure"},
      {id: 16, name: "Animation"}
    ]; 
    const selected = [12, 16];
    const onChange = jest.fn();
    const component = <Checklist list={genres} checked={selected} onChange={onChange} />
    const tree = renderer.create(component).toJSON();

    it('should render', () => {
      expect(tree)
        .toMatchSnapshot();
    });

    it('should contain a list of items', () => {
      const numChildren = tree.children.length;
      expect(numChildren)
        .toBeTruthy();
      expect(numChildren)
        .toEqual(genres.length);
    });
  });

});