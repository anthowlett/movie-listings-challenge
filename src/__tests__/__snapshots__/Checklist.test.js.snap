// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filters <Checklist /> should render 1`] = `
.c0 input {
  opacity: 0;
  position: absolute;
}

.c0 label {
  display: inline-block;
  position: relative;
  margin: 0 5px 10px;
  padding: 7px;
  vertical-align: middle;
  font-size: 14px;
  cursor: pointer;
  border: 1px solid black;
  -webkit-transition: background-color 0.25s;
  transition: background-color 0.25s;
  border-radius: 5px;
}

.c0 input:checked + label {
  color: white;
  background-color: #00A14D;
  border-color: green;
}

<div>
  <span
    className="c0"
  >
    <input
      checked={false}
      id={28}
      name="list"
      onChange={[Function]}
      type="checkbox"
      value={28}
    />
    <label
      htmlFor={28}
    >
      Action
    </label>
  </span>
  <span
    className="c0"
  >
    <input
      checked={true}
      id={12}
      name="list"
      onChange={[Function]}
      type="checkbox"
      value={12}
    />
    <label
      htmlFor={12}
    >
      Adventure
    </label>
  </span>
  <span
    className="c0"
  >
    <input
      checked={true}
      id={16}
      name="list"
      onChange={[Function]}
      type="checkbox"
      value={16}
    />
    <label
      htmlFor={16}
    >
      Animation
    </label>
  </span>
</div>
`;
